bigbone/src/main/kotlin/social/bigbone/api/entity/Instance.kt

```diff
@@ -85,9 +85,214 @@ data class Instance(
+            val maxMediaAttachments: Int = 0,
+
+            /**
+             * Each URL in a status will be assumed to be exactly this many characters.
+             */
+            @SerializedName("characters_reserved_per_url")
+            val charactersReservedPerUrl: Int = 0
+        )
+
+        /**
+         * Hints for which attachments will be accepted.
+         * @see <a href="https://docs.joinmastodon.org/entities/V1_Instance/">Mastodon API V1::Instance</a>
+         */
+        data class MediaAttachments(
+            /**
+             * Contains MIME types that can be uploaded.
+             */
+            @SerializedName("supported_mime_types")
+            val supportedMimeTypes: List<String> = emptyList(),
+
+            /**
+             * The maximum size of any uploaded image, in bytes.
+             */
+            @SerializedName("image_size_limit")
+            val imageSizeLimit: Int = 0,
+
+            /**
+             * The maximum number of pixels (width times height) for image uploads.
+             */
+            @SerializedName("image_matrix_limit")
+            val imageMatrixLimit: Int = 0,
+
+            /**
+             * The maximum size of any uploaded video, in bytes.
+             */
+            @SerializedName("video_size_limit")
+            val videoSizeLimit: Int = 0,
+
+            /**
+             * The maximum frame rate for any uploaded video.
+             */
+            @SerializedName("video_frame_rate_limit")
+            val videoFrameRateLimit: Int = 0,
+
+            /**
+             * The maximum number of pixels (width times height) for video uploads.
+             */
+            @SerializedName("video_matrix_limit")
+            val videoMatrixLimit: Int = 0
+        )
+
+        /**
+         * Limits related to polls.
+         * @see <a href="https://docs.joinmastodon.org/entities/V1_Instance/">Mastodon API V1::Instance</a>
+         */
+        data class Polls(
+            /**
+             * Each poll is allowed to have up to this many options.
+             */
+            @SerializedName("max_options")
+            val maxOptions: Int = 0,
+
+            /**
+             * Each poll option is allowed to have this many characters.
+             */
+            @SerializedName("max_characters_per_option")
+            val maxCharactersPerOption: Int = 0,
+
+            /**
+             * The shortest allowed poll duration, in seconds.
+             */
+            @SerializedName("min_expiration")
+            val minExpiration: Int = 0,
+
+            /**
+             * The longest allowed poll duration, in seconds.
+             */
+            @SerializedName("max_expiration")
+            val maxExpiration: Int = 0
+        )
+    }
+
+    /**
+     * Represents a rule that server users should follow.
+     * @see <a href="https://docs.joinmastodon.org/entities/Rule/">Mastodon API Rule</a>
+     */
+    data class Rule(
```

Comment:
This seems to be a first-level entity according to Mastodon documentation. Could be moved to its own file, especially if it is used elsewhere.

Reply:
Thx, fixed!